python/cudf/benchmarks/common/utils.py

```diff
@@ -0,0 +1,268 @@
+            f"only exist for the values {', '.join(c for c in NUM_ROWS)}"
+        )
+        row_str = f"_rows_{rows}"
+
+    fixture_name = f"{cls}{dtype_str}{null_str}{col_str}{row_str}"
+
+    def deco(bm):
+        # pytests's test collection process relies on parsing the globals dict
+        # to find test functions and identify their parameters for the purpose
+        # of fixtures and parameters. Therefore, the primary purpose of this
+        # decorator is to define a new benchmark function with a signature
+        # identical to that of the decorated benchmark except with the user's
+        # fixture name replaced by the true fixture name based on the arguments
+        # to accepts_cudf_fixture.
+        parameters = inspect.signature(bm).parameters
+
+        # Note: This logic assumes that any benchmark using this fixture has at
+        # least two parameters since they must be using both the
+        # pytest-benchmark `benchmark` fixture and the cudf object.
+        params_str = ", ".join(f"{p}" for p in parameters if p != name)
+        arg_str = ", ".join(f"{p}={p}" for p in parameters if p != name)
+
+        if params_str:
+            params_str += ", "
+        if arg_str:
+            arg_str += ", "
+
+        params_str += f"{fixture_name}"
+        arg_str += f"{name}={fixture_name}"
+
+        src = textwrap.dedent(
+            f"""
+            def wrapped_bm({params_str}):
+                return bm({arg_str})
+            """
+        )
+        globals_ = {"bm": bm}
+        exec(src, globals_)
+        wrapped_bm = globals_["wrapped_bm"]
+        # In case marks were applied to the original benchmark, copy them over.
+        if marks := getattr(bm, "pytestmark", None):
+            wrapped_bm.pytestmark = marks
+        return wrapped_bm
+
+    return deco
+
+
+class OrderedSet(MutableSet):
+    """A minimal OrderedSet implementation built on a dict.
+
+    This implementation exploits the fact that dicts are ordered as of Python
+    3.7. It is not intended to be performant, so only the minimal set of
+    methods are implemented. We need this class to ensure that fixture names
+    are constructed deterministically, otherwise pytest-xdist will complain if
+    different threads have seemingly different tests.
+    """
+
+    def __init__(self, args=None):
+        args = args or []
+        self._data = {value: None for value in args}
+
+    def __contains__(self, key):
+        return key in self._data
+
+    def __iter__(self):
+        return iter(self._data)
+
+    def __len__(self):
+        return len(self._data)
+
+    def __repr__(self):
+        # Helpful for debugging.
+        data = ", ".join(str(i) for i in self._data)
+        return f"{self.__class__.__name__}({data})"
+
+    def add(self, value):
+        self._data[value] = None
+
+    def discard(self, value):
+        self._data.pop(value, None)
+
+
+def make_fixture(name, func, globals_, fixtures):
```

Comment:
Perhaps `make_fixture_and_inject_to_globals` is more descriptive?

Reply:
It is more descriptive, but I think that's more detail than is really needed. This function is only two lines, and at that point the function name is almost as long as just copy-pasting the body of the function :) If someone really needs to understand the globals component they can read it.